sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/TransactionalBatchOperationResult.java

```diff
@@ -0,0 +1,143 @@
+        this.eTag = other.eTag;
+        this.requestCharge = other.requestCharge;
+        this.retryAfter = other.retryAfter;
+        this.resourceObject = other.resourceObject;
+    }
+
+    /**
+     * Instantiates a new Transactional batch operation result using a TransactionalBatchOperationResult and item.
+     *
+     * @param result the TransactionalBatchOperationResult.
+     * @param item the item.
+     */
+    TransactionalBatchOperationResult(TransactionalBatchOperationResult<?> result, TResource item) {
+        this(result);
+        this.item = item;
+    }
+
+    /**
+     * Initializes a new instance of the {@link TransactionalBatchOperationResult} class.
+     */
+    TransactionalBatchOperationResult(String eTag,
+                                      Double requestCharge,
+                                      ObjectNode resourceObject,
+                                      int responseStatus,
+                                      Duration retryAfter,
+                                      Integer subStatusCode) {
+        checkNotNull(responseStatus, "expected non-null responseStatus");
+
+        this.eTag = eTag;
+        this.requestCharge = requestCharge;
+        this.resourceObject = resourceObject;
+        this.responseStatus = responseStatus;
+        this.retryAfter = retryAfter;
+        this.subStatusCode = subStatusCode;
+    }
+
+    /**
+     * Gets the entity tag associated with the current item.
+     *
+     * ETags are used for concurrency checking when updating resources.
+     *
+     * @return Entity tag associated with the current item.
+     */
+    public String getETag() {
+        return this.eTag;
+    }
+
+    /**
+     * Gets the request charge in request units for the current operation.
+     *
+     * @return Request charge in request units for the current operation.
+     */
+    public Double getRequestCharge() {
+        return this.requestCharge;
+    }
+
+    /**
+     * Gets the item associated with the current result.
+     *
+     * @return item associated with the current result.
+     */
+    public TResource getItem() {
+        return this.item;
+    }
+
+    /**
+     * Gets retry after.
+     *
+     * @return the retry after
+     */
+    public Duration getRetryAfter() {
+        return this.retryAfter;
+    }
+
+    /**
+     * Gets sub status code.
+     *
+     * @return the sub status code
+     */
+    public Integer getSubStatusCode() {
+        return this.subStatusCode;
+    }
+
+    /**
+     * Gets a value indicating whether the current operation completed successfully.
+     *
+     * @return {@code true} if the current operation completed successfully; {@code false} otherwise.
+     */
+    public boolean isSuccessStatusCode() {
+        return 200 <= this.responseStatus && this.responseStatus <= 299;
+    }
+
+    /**
+     * Gets response status.
+     *
+     * @return the response status
+     */
+    public int getResponseStatus() {
```

Comment:
the method names should match the relevant methods of `CosmosItemResponse`

we have `CosmosItemResponse.getStatusCode()` any reason this one is named differently?

Reply:
No. Will change it. Thanks!